libbeat/esleg/eslegclient/connection.go

```diff
@@ -452,17 +459,18 @@ func (conn *Connection) execHTTPRequest(req *http.Request) (int, []byte, error)
 	defer closing(resp.Body, conn.log)
 
 	status := resp.StatusCode
-	obj, err := ioutil.ReadAll(resp.Body)
+	conn.responseBuffer.Reset()
+	_, err = io.Copy(conn.responseBuffer, resp.Body)
```

Comment:
@alexsapran I think we should not re-use the buffer here because the size is not static. This buffer will never release its allocated memory and the size of this memory will be equal to the size of the biggest ever processed HTTP response. I'd rather prefer to release the memory and re-allocate smaller buffers when needed.

I think we should switch it in favour of https://github.com/elastic/elastic-agent-libs/pull/183 and compare performance. This implementation should be the most optimized one.

Reply:
Yes, you are correct @rdner to the use case you mention, but this might be just fine in this particular scenario because the output is relational to the batching we do. 

So the response size is within a ballpark the same for each bulk we publish. 
We can adjust this and make it so that we only allocate a buffer just for the consumption of that response.
Since we are in the process of evaluating other usages of the `ioutil.ReadAll` it might make sense to adjust this as well.